proof/proof.go

```diff
@@ -446,3 +446,29 @@ func (p *Proof) IsUnknownVersion() bool {
 		return true
 	}
 }
+
+// SparseDecode can be used to decode a proof from a reader without decoding
+// and parsing the entire thing. This handles ignoring the magic bytes, and
+// will decode directly into the target records.
+func SparseDecode(r io.Reader, records ...tlv.Record) error {
+	// The very first byte of the serialized proof is a magic byte, so
```

Comment:
Why ignore magic bytes here? IIUC it costs us little to actually validate them here.


Or is it unneeded b/c internal use only.

Reply:
Later commit ends up validating it, might have been a fixup mixup.